src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicy.cs

```diff
@@ -56,6 +56,25 @@ private static IEnumerable<PropertyInfo> GetPropertiesRecursive(Type type)
             var unseenProperties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
                 .Where(p => p.CanRead && p.GetMethod.IsPublic && p.GetIndexParameters().Length == 0 && !seenNames.Contains(p.Name));
 
+            if (_options.RespectMetadataTypeAttribute)
+            {
+                var metaProp = new List<PropertyInfo>();
+                // find Metadata Class
+                // Take only first Entry, metadatatypeAttribute definition specifies AllowMultiple=false
+                // see https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.metadatatypeattribute?view=net-9.0#definition 
+                var metaDataType = type.GetCustomAttributes(true).Where(t => t.GetType().FullName == "System.ComponentModel.DataAnnotations.MetadataTypeAttribute").FirstOrDefault();
+                if (metaDataType != null)
+                {
+                    var metaClass = (Type)metaDataType.GetType().GetProperty("MetadataClassType").GetValue(metaDataType, null);
+                    // find all properties with Custom Attributes which are in referenced class
+                    metaProp = metaClass.GetProperties().Where(mp => mp.CustomAttributes.Count() > 0 && unseenProperties.Any(up => up.Name == mp.Name)).ToList();
+                    // replace all found properties in unseenProperties with those from Metadataclass
+                    var removedAttr = unseenProperties.Where(up => metaProp.Any(mp => mp.Name != up.Name)).ToList();
+                    removedAttr.AddRange(metaProp);
+                    unseenProperties = removedAttr;
```

Comment:
It is wrong. You should not replace original properties with properties from another type. They are physically different things. Now I understand what you meant when you talked about the impossibility of obtaining values. Instead you should somehow track new properties along with "real" properties using attributes from new properties but value - from "real".

Reply:
OK. I see basically two options:

1.  Implement an extension method to get the Custom Attributes from Metadataclass and use this method in subsequent code. It should deliver the custom attributes from either data call or metadatatype class if present. 
2.  Implement a sort of CustomPropertyInfo class which inherits PropertyInfo from data class and has a property MetadataCustomAttributes from Metaclass. If there is no metadataclass this property is null. 

What would you prefer?

Reply:
I would move further with (2) since it looks less destructive to existing code.

Reply:
OK after some research I decided to go for an extension Method. Much simpler to implement and from my perspective much less intrusive.